### PR TITLE
Fix missing padding of the transformed 2d-src and 1d-dst tensors in Generic-Tensor Reduction (SWDEV-284291)

### PR DESCRIPTION
**Pull request by qianfengz**
This is pushed in order to solve the issue raised by [SWDEV-284291](https://ontrack-internal.amd.com/browse/SWDEV-284291)

Regular testing has passed on (rocm-3.10,rocm-3.7,rocm-4.2)/MI00,  rocm-4.2/MI25, rocm-4.2/Navi21, rocm-3.10/MI60 

Method for regular testing
`#>bin/test_reduce_test --all`
`#>bin/test_reduce_test --half --all`

### CODE REVIEW

**Comment by shaojiewang** [comment]
Is the issue fixed in Huggingface as described in SWDEV-284291? Could you please attach some test log?

**Comment by asroy** [comment]
@qianfengz Please confirm if this PR would fix the issue for all these files

https://github.com/ROCmSoftwarePlatform/MIOpen/blob/develop/src/kernels/composable_kernel/include/kernel_algorithm/gridwise_generic_2d_reduction_blockwise.hpp
https://github.com/ROCmSoftwarePlatform/MIOpen/blob/develop/src/kernels/composable_kernel/include/kernel_algorithm/gridwise_generic_2d_reduction_direct_threadwise.hpp
https://github.com/ROCmSoftwarePlatform/MIOpen/blob/develop/src/kernels/composable_kernel/include/kernel_algorithm/gridwise_generic_2d_reduction_direct_warpwise.hpp
https://github.com/ROCmSoftwarePlatform/MIOpen/blob/develop/src/kernels/composable_kernel/include/kernel_algorithm/gridwise_generic_2d_reduction_multiblock.hpp

**Comment by qianfengz** [comment]
> @qianfengz Please confirm if this PR would fix the issue for all these files
> 
> https://github.com/ROCmSoftwarePlatform/MIOpen/blob/develop/src/kernels/composable_kernel/include/kernel_algorithm/gridwise_generic_2d_reduction_blockwise.hpp
> https://github.com/ROCmSoftwarePlatform/MIOpen/blob/develop/src/kernels/composable_kernel/include/kernel_algorithm/gridwise_generic_2d_reduction_direct_threadwise.hpp
> https://github.com/ROCmSoftwarePlatform/MIOpen/blob/develop/src/kernels/composable_kernel/include/kernel_algorithm/gridwise_generic_2d_reduction_direct_warpwise.hpp
> https://github.com/ROCmSoftwarePlatform/MIOpen/blob/develop/src/kernels/composable_kernel/include/kernel_algorithm/gridwise_generic_2d_reduction_multiblock.hpp

Tensor transform is completely done in 
  ` src/kernels/composable_kernel/include/kernel_algorithm/gridwise_generic_reduction.hpp`
  before calling the functions in those files

**Comment by junliume** [comment]
@atamazov @asroy this is a release cherry picking candidate, can you please review? Looks like CI has passed: http://micimaster.amd.com/blue/organizations/jenkins/MLLibs%2FMIOpen/detail/reduction-oob/1/pipeline/

File: test/reduce_test.cpp [hunk 1]
```diff
@@ -41,10 +41,6 @@
 
 #include "cpu_reduce_util.hpp"
 
-/// Not reproducible with ROCm 4.0 and 4.1.
```
Comment (atamazov) [inline test/reduce_test.cpp:44]: [Note] You can also remove config.h at line 26, it is not necessary anymore. See https://github.com/ROCmSoftwarePlatform/MIOpen/pull/774/commits/58831650790227e9fc6f84c9365fd9a66c413ac7

**Review by atamazov — APPROVED** [review]
LGTM

**Pull request merged**